jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -191,6 +191,7 @@ public BaseImageWithAuthorization call()
                 buildConfiguration.getBaseImageConfiguration().getImageRepository())
             .setAllowInsecureRegistries(buildConfiguration.getAllowInsecureRegistries())
             .setAuthorization(registryAuthorization)
+            .setUserAgentSuffix(buildConfiguration.getToolName())
```

Comment:
Hmm, it might be better to have a centralized location where this is set so that it doesn't need to be called for every time we create a new `RegistryClient`. I'm thinking perhaps something at the `BuildConfiguration` level (the class itself or some other class that uses it) could have a `newRegistryClientFactory` that generates a new `RegistryClient.Factory` with the user agent suffix (and other fields) pre-populated so that the only methods that need to be called are `setAuthorization` and `newRegistryClient`.